### PR TITLE
Add CHECK for cobalt::loader::FetcherCache

### PR DESCRIPTION
**Pull request by borongc**
Add more CHECK to identify the root cause of the crash.

b/270993319

### CODE REVIEW

File: cobalt/loader/fetcher_cache.cc [hunk 1]
```diff
@@ -348,6 +348,9 @@ void FetcherCache::OnFetchSuccess(
 
   total_size_ += entry->capacity();
   ++count_resources_cached_;
+  // TODO(b/270993319): For debugging cache integrity issues in production
+  //                    only, remove after identifying the root cause.
+  CHECK_EQ(count_resources_cached_, (int)cache_entries_.size());
```
Comment (xiaomings) [inline cobalt/loader/fetcher_cache.cc:353]: `static_cast<int>(cache_entries_.size())`.  We generally avoid c-style casts.

Comment (borongc) [reply cobalt/loader/fetcher_cache.cc:353]: Fixed.

**Review by xiaomings — APPROVED** [review]

**Comment by oxve** [comment]
@borongc Next time you use squash and merge, can you make sure the bug id is in the squashed commit message?

It looks like it was removed before merging https://github.com/youtube/cobalt/actions/runs/6881357377

**Comment by borongc** [comment]
@oxve Will check next time. Would this be fixable for this merge?

**Pull request merged**